integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py

```diff
@@ -288,3 +190,170 @@ def from_dict(cls, data: Dict[str, Any]) -> "AmazonBedrockChatGenerator":
             ["aws_access_key_id", "aws_secret_access_key", "aws_session_token", "aws_region_name", "aws_profile_name"],
         )
         return default_from_dict(cls, data)
+
+    @component.output_types(replies=List[ChatMessage])
+    def run(
+        self,
+        messages: List[ChatMessage],
+        streaming_callback: Optional[Callable[[StreamingChunk], None]] = None,
+        generation_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        generation_kwargs = generation_kwargs or {}
+
+        # Merge generation_kwargs with defaults
+        merged_kwargs = self.generation_kwargs.copy()
+        merged_kwargs.update(generation_kwargs)
+
+        # Extract known inference parameters
+        # See https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_InferenceConfiguration.html
+        inference_config = {
+            key: merged_kwargs.pop(key, None)
+            for key in ["maxTokens", "stopSequences", "temperature", "topP"]
+            if key in merged_kwargs
+        }
+
+        # Extract tool configuration if present
+        # See https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_ToolConfiguration.html
+        tool_config = merged_kwargs.pop("toolConfig", None)
+
+        # Any remaining kwargs go to additionalModelRequestFields
+        additional_fields = merged_kwargs if merged_kwargs else None
+
+        # Prepare system prompts and messages
+        system_prompts = []
+        if messages and messages[0].is_from(ChatRole.SYSTEM):
+            system_prompts = [{"text": messages[0].content}]
+            messages = messages[1:]
+
+        messages_list = [{"role": msg.role.value, "content": [{"text": msg.content}]} for msg in messages]
+
+        try:
```

Comment:
we can move `try` closer to the actual client usage
(after `callback = streaming_callback or self.streaming_callback`)